photon-ml/src/main/scala/com/linkedin/photon/ml/FeatureIndexingJob.scala

```diff
@@ -19,79 +19,100 @@ import org.apache.hadoop.conf.Configuration
-
+import scopt.OptionParser
 
 import java.util.{List => JList}
 
 /**
-  * A class that builds feature index map as an independent Spark job. Recommended when feature space is large,
+  * A class that builds feature index map as an independent Spark job. Recommended when the feature space is large,
   * typically when there are more than 200k unique features.
   *
-  * The job expected three required arguments and one optional one:
-  * [input_data_path]: The input path of data
-  * [partition_num]: The number of partitions to break the storage into. This is merely introduced as an optimization at
-  *   build stage that we don't necessary have to shuffle all features into the same partition but rather we could
-  *   just hold them into multiple ones. A heuristically good number could be 1-10, or even as large as 20; but please
-  *   avoid setting it into an arbitrarily large number. Use just 1 or 2 if you find the indexing job is already fast
-  *   enough.
-  * [output_dir]: The output directory
-  * [if_add_intercept] (optional, default=true): whether include INTERCEPT into the map.
+  * The job expects three required arguments and three optional ones:
   *
+  * input-paths: The input paths of the data
+  * partition-num: The number of partitions to break the storage into. This is merely introduced as an optimization at
+  *   the index building stage, so that we don't need to shuffle all features into the same partition. A heuristically
+  *   good number would be 1-10, or even as large as 20; but please avoid setting it into an arbitrarily large number.
+  *   Usejust 1 or 2 if you find the indexing job is already fast enough.
+  * output-path: The output path
+  * date-range: Date range for the input data represented in the form start.date-end.date, e.g. 20150501-20150631
+  * date-range-days-ago: Date range for the input data represented in the form start.daysAgo-end.daysAgo, e.g. 90-1
+  * add-intercept: whether include INTERCEPT in the map (optional, default: true).
+  * data-format: Input data format. (optional, default: TRAINING_EXAMPLE, options:
+  *   [TRAINING_EXAMPLE|RESPONSE_PREDICTION])
+  * feature-shard-id-to-feature-section-keys-map: A map between the feature shard id and it's corresponding feature
+  *   section keys, in the following format: shardId1:sectionKey1,sectionKey2|shardId2:sectionKey2,sectionKey3.
   */
-class FeatureIndexingJob(val sc: SparkContext,
-                         val inputPath: String,
-                         val partitionNum: Int,
-                         val outputDir: String,
-                         val addIntercept: Boolean,
-                         val fieldNames: FieldNames) {
-  private val outputPath: Path = {
-    val path = new Path(outputDir)
-    val fs = FileSystem.get(new Configuration())
+class FeatureIndexingJob(
+    val sc: SparkContext,
+    val inputPaths: Seq[String],
+    val partitionNum: Int,
+    val outputPath: String,
+    val addIntercept: Boolean,
```

Comment:
Can you double check the way we deal with intercepts and the impact of using the global flag ```addIntercept``` instead of a per-shard one?

Line 182 (before this PR) in ```DataProcessingUtils.scala```:
```val isAddingInterceptToFeatureMap = nameAndTermToIndexMap.contains(NameAndTerm.INTERCEPT_NAME_AND_TERM)```
So it loos like we are deciding whether to add the intercept based on whether the intercept term is added to the feature map or not.

Are we keeping the same logic as above after this PR?